### PR TITLE
Removed InputCutInfo, disabled input cut in OVC.

### PR DESCRIPTION
**Pull request by popovaan**
Description: 

- Removed InputCutInfo.
- Disabled input/output cut in OVC. In case if node name specified that is not among model inputs exception is raised.
- Fixed logic of output_model parameter. If directory is specified, output_model works as output_dir.
- Error massages for frontend errors (like Translator not found) are made smaller in OVC tool.
- Separated OVC/MO Python API tests.

Ticket: 115752

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

File: tools/ovc/openvino/tools/ovc/moc_frontend/pipeline.py [hunk 1]
```diff
@@ -109,10 +119,17 @@ def add_names_to_tensors(model: InputModel, places: List[Place]):
     model_inputs = input_model.get_inputs()
     inputs_equal = True
     if user_shapes:
+
+        # TODO: Remove this line when new 'cut' helper is introduced
+        raise_exception_for_input_output_cut(model_inputs, user_shapes, True)
```
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/moc_frontend/pipeline.py:124]: Are we sure that `model_inputs` is correct for all FEs (even for TF FE for different formats)?

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py [hunk 2]
```diff
@@ -35,132 +35,23 @@ def basic(self, input_model, argv_input, inputs, dtype, expected, only_conversio
...
+        #     ),
+        #     (
+        #             "add:0{i32}[3],z{i32}",
+        #             {"add:0": np.array([4, 5, 6], dtype=np.int32), "z": np.array([1, 2, 3], dtype=np.int32)},
+        #             np.array([4, 10, 18], dtype=np.int32),
+        #             np.int32,
+        #             None
+        #     ),
+        # ],
     )
```
Comment (rkazants) [inline tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:55]: please add test to check that exception about cutting is generated correctly

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py [hunk 3]
```diff
@@ -169,20 +60,21 @@ def test_cutting_fp32(self, input_freezing_value, inputs, expected,
```
Comment (rkazants) [inline tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:60]: Now it seems no tests remains here

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py [hunk 4]
```diff
@@ -169,20 +60,21 @@ def test_cutting_fp32(self, input_freezing_value, inputs, expected,
...
-            (
-                    "x,y",
-                    {"x": np.array([[-3, 20, 1]], dtype=np.int32), "y": np.array([[10, -11, -17]], dtype=np.int32)},
-                    np.array([[7, 9, -16]], dtype=np.int32),
-                    np.int32,
-                    None
-            ),
+            # TODO: Return this test when new 'cut' helper is introduced
+            # (
+            #         "x[1,4],y[4]",
```
Comment (rkazants) [inline tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:65]: why do we need cut helper if there is no cutting?

Comment (rkazants) [inline tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:65]: why do we need cut helper if there is no cutting?

File: tools/ovc/openvino/tools/ovc/convert.py [hunk 5]
```diff
@@ -63,27 +58,18 @@ def convert_model(
...
+            Input can be set by a list of tuples or a dictionary. Each tuple can contain optionally input name (string),
+            input type (ov.Type) or input shape (ov.Shape, ov.PartialShape, list, tuple).
+            Example: input=("op_name", PartialShape([-1, 3, 100, 100]), ov.Type.f32).
+            Alternatively input can be set by a dictionary, where key - input name,
+            value - tuple with input parameters (shape or type).
+            Example: input={"op_name_1": ([1, 2, 3], ov.Type.f32), "op_name_2": ov.Type.i32}
+            The order of inputs in converted model will match the order of specified inputs.
+            If data type is not specified explicitly then there are two options: if
+            input node is a parameter, data type is taken from the original node data type,
+            if input node is not a parameter, data type is set to f32.
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/convert.py:71]: Yes, we should probably delete it.

File: tools/ovc/openvino/tools/ovc/help.py [hunk 6]
```diff
@@ -13,65 +13,21 @@ def get_convert_model_help_specifics():
...
-                 'semicolon. Examples: \"--transform LowLatency2\" or \"--'
-                 'transform Pruning" or "--transform '
-                 'LowLatency2[use_const_initializer=False]" or "--'
-                 'transform "MakeStateful[param_res_names= {\'input_name_1\':'
-                 '\'output_name_1\',\'input_name_2\':\'output_name_2\'}]\" \n'
-                 'Available transformations: "LowLatency2", "MakeStateful", "Pruning"'},
+                 'original node data type, if input node is not a parameter, '
+                 'data type is set to f32. Example, to set `input_1` input '
+                 'with shape [1,100] and float32 type, and `sequence_len` input '
+                 'with int32 type \"input_1[1,100]{f32},sequence_len{i32}\".'},
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/help.py:30]: In help.py description provided only for parameters which description is different from convert_model(). 
`output` description is copied from docstring when cli parser is created.

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py (hunk 3 above)
Comment (popovaan) [reply tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:60]: It is planned to return input/output cut functionality when "cut" helper is created. So I left the tests here commented, so we could easily return them later.
But I can also remove this file to make code cleaner.

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py (hunk 4 above)
Comment (popovaan) [reply tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:65]: No cutting is temporal behavior. In future we plan to add output/input cut using new `cut` helper.

Comment (popovaan) [reply tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:65]: Answered above.

File: tools/openvino_dev/src/openvino/__init__.py [hunk 7]
```diff
@@ -57,6 +57,6 @@
 # Tools
 try:
     # Model Conversion API - ovc should reside in the main namespace
-    from openvino.tools.ovc import convert_model, InputCutInfo
+    from openvino.tools.ovc import convert_model
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/openvino_dev/src/openvino/__init__.py:60]: In this line `InputCutInfo` is removed from `openvino` namespace.
`InputCutInfo` still exists in openvino.tools.mo.
You can check: 
tests/layer_tests/mo_python_api_tests/test_mo_convert_complex_params.py
`InputCutInfo` is imported there from openvino.tools.mo and tested.

File: tools/mo/openvino/__init__.py [hunk 8]
```diff
@@ -57,6 +57,6 @@
 # Tools
 try:
     # Model Conversion API - ovc should reside in the main namespace
-    from openvino.tools.ovc import convert_model, InputCutInfo
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/mo/openvino/__init__.py:60]: In this line `InputCutInfo` is removed from `openvino` namespace.
`InputCutInfo` still exists in openvino.tools.mo.
You can check: 
tests/layer_tests/mo_python_api_tests/test_mo_convert_complex_params.py
`InputCutInfo` is imported there from openvino.tools.mo and tested.

File: tools/benchmark_tool/openvino/__init__.py [hunk 9]
```diff
@@ -57,6 +57,6 @@
 # Tools
 try:
     # Model Conversion API - ovc should reside in the main namespace
-    from openvino.tools.ovc import convert_model, InputCutInfo
+    from openvino.tools.ovc import convert_model
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/benchmark_tool/openvino/__init__.py:60]: openvino root init file need to be the same in all sub packages which use openvino namespace. 
Please check this PR where all imports from openvino.runtime are moved to openvino namespace https://github.com/openvinotoolkit/openvino/pull/18157

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py (hunk 4 above)
Comment (rkazants) [reply tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:65]: I am afraid that we have no test remained with real inputs and outpus specified. Looks only OOB conversion without real inputs and output specification

File: tools/ovc/openvino/tools/ovc/cli_parser.py [hunk 10]
```diff
@@ -3,27 +3,27 @@
...
 
 import openvino
 from openvino.tools.ovc.convert_data_type import destination_type_to_np_data_type
 from openvino.tools.ovc.error import Error
-from openvino.tools.ovc.utils import get_mo_root_dir
 from openvino.tools.ovc.help import get_convert_model_help_specifics
+from openvino.tools.ovc.utils import get_mo_root_dir
+
+# Helper class for storing input cut information
+_InputCutInfo = namedtuple("InputCutInfo", ["name", "shape", "type", "value"], defaults=[None, None, None, None])
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:26]: `InputCutInfo` was used internally by MO/OVC to store parsed input values. 
As `InputCutInfo` was removed from public API I added internal helper class for internal usage.

File: tools/ovc/openvino/tools/ovc/cli_parser.py [hunk 11]
```diff
@@ -85,19 +77,18 @@ def single_input_to_input_cut_info(input: [str, tuple, list, PartialShape, Type,
...
 
     # We don't expect here single unnamed value. If list of int is set it is considered as shape.
     # Setting of value is expected only using InputCutInfo or string analog.
 
-    raise Exception("Unexpected object provided for input. Expected openvino.tools.ovc.InputCutInfo "
-                    "or tuple or str. Got {}".format(type(input)))
+    raise Exception("Unexpected object provided for input. Expected tuple, Shape, PartialShape, Type or str. Got {}".format(type(input)))
 
 
 def input_to_input_cut_info(input: [str, tuple, list]):
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:94]: This method is used for params parsing. `InputCutInfo` is used here as structure for storing parsed values. In this PR I replace it with internal class `_InputCutInfo`.

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py (hunk 4 above)
Comment (rkazants) [reply tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:65]: there is no cutting in this test and this test should work via ovc

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_pytorch.py [hunk 12]
```diff
@@ -255,7 +254,7 @@ def create_pytorch_nn_module_sample_input_int32_two_inputs(tmp_dir):
     ref_model = make_ref_pt_model_two_inputs(
         [PartialShape([-1, 3, -1, -1]), inp_shapes[1]], dtype=np.int32)
 
-    return pt_model, ref_model, {'input': [(np.int32, inp_shapes[0]), (np.int32, inp_shapes[1])],
+    return pt_model, ref_model, {'input': [(ov.Type.i32, inp_shapes[0]), (ov.Type.i32, inp_shapes[1])],
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_pytorch.py]: After discussion with Sergey, he agreed to return numpy type. Done.

File: tools/ovc/openvino/tools/ovc/convert.py [hunk 13]
```diff
@@ -63,27 +58,18 @@ def convert_model(
...
-            `input_1` with shape [1,100], and Parameter node `sequence_len` with
-            scalar input with value `150`, and boolean input `is_training` with
-            `False` value use the following format: "input_1[1,100],sequence_len->150,is_training->False".
-            Another example, use the following format to set input port 0 of the node
-            `node_name1` with the shape [3,4] as an input node and freeze output
-            port 1 of the node `node_name2` with the value [20,15] of the int32 type
-            and shape [2]: "0:node_name1[3,4],node_name2:1[2]{i32}->[20,15]".
+            Information of model input required for model conversion.
+            Input can be set by a list of tuples or a dictionary. Each tuple can contain optionally input name (string),
+            input type (ov.Type) or input shape (ov.Shape, ov.PartialShape, list, tuple).
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/convert.py]: Returned numpy type.

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py (hunk 4 above)
Comment (popovaan) [reply tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:65]: I checked the model, it has `tf.placeholder()` with name `x` and `tf.placeholder_with_default()` with name `y`. `tf.placeholder_with_default()` becomes constant after `input_model = moc_front_end.load(...)`. 
`input_model.get_inputs()` returns only one input with name 'x'.
The further logic applies usual input cut.

It looks like, to make this case work we need to check inputs of original TF model. But in this case the code will be FW specific and we will need to add this logic for each FW.

File: tools/ovc/openvino/tools/ovc/convert.py [hunk 14]
```diff
@@ -63,27 +58,18 @@ def convert_model(
...
+            Example: input=("op_name", PartialShape([-1, 3, 100, 100]), ov.Type.f32).
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/convert.py:64]: Added examples with list of tuples and without names. Please check.

File: tools/ovc/openvino/tools/ovc/convert_impl.py [hunk 15]
```diff
@@ -80,10 +79,11 @@ def print_argv(argv: argparse.Namespace, model_name: str):
 def arguments_post_parsing(argv: argparse.Namespace):
     # TODO: This function looks similar to another one. Check for code duplicates.
     log.debug("Model Conversion API started")
-    log.debug('Output model name would be {}{{.xml, .bin}}'.format(argv.output_model))
+    if not argv.is_python_api_used:
+        log.debug('Output model name would be {}{{.xml, .bin}}'.format(argv.output_model))
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/convert_impl.py:83]: In this case `argv.output_model` works the same as `argv.output_dir` in MO. Added test for directory in `output_model` in `ovc_python_api_tests/test_ovc_cli_tool.py`

File: tools/ovc/openvino/tools/ovc/help.py [hunk 16]
```diff
@@ -13,65 +13,21 @@ def get_convert_model_help_specifics():
              'action': CanonicalizePathCheckExistenceAction,
```
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/help.py:13]: Done.

File: tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py (hunk 2 above)
Comment (popovaan) [reply tools/ovc/unit_tests/moc_tf_fe/conversion_basic_models_test.py:55]: Done. The test is in ovc_python_api_tests/test_complex_params.py group.

**Review by slyalin — APPROVED** [review]

File: tools/ovc/openvino/tools/ovc/cli_parser.py [hunk 17]
```diff
@@ -1033,7 +957,7 @@ def check_bool(value):
 
 def depersonalize(value: str, key: str):
     dir_keys = [
-        'output_dir', 'extension', 'saved_model_dir', 'tensorboard_logdir', 'caffe_parser_path'
+        'extension', 'saved_model_dir', 'tensorboard_logdir', 'caffe_parser_path'
```
Comment (rkazants) [inline tools/ovc/openvino/tools/ovc/cli_parser.py:960]: why do we still mention 'saved_model_dir', 'tensorboard_logdir', 'caffe_parser_path'?

File: tools/ovc/openvino/tools/ovc/convert.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (rkazants) [reply tools/ovc/openvino/tools/ovc/convert.py:73]: Please delete

File: tools/ovc/openvino/tools/ovc/cli_parser.py (hunk 17 above)
Comment (rkazants) [reply tools/ovc/openvino/tools/ovc/cli_parser.py:960]: Please check in other files by key words like caffe, mxnet, kaldi, tensorboard, etc.

File: tools/ovc/openvino/tools/ovc/convert.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/convert.py]: Done.

File: tools/ovc/openvino/tools/ovc/cli_parser.py (hunk 17 above)
Comment (popovaan) [reply tools/ovc/openvino/tools/ovc/cli_parser.py]: Removed 'saved_model_dir', 'tensorboard_logdir', 'caffe_parser_path' from this list.
Checked caffe, mxnet, kaldi, tensorboard in other files, no other mentions were found.

**Review by rkazants — APPROVED** [review]

**Pull request merged**